lighthouse-core/gather/gatherers/dobetterweb/document-event-listeners.js

```diff
@@ -93,17 +106,16 @@ class PageLevelEventListeners extends Gatherer {
   }
 
   /**
-   * Aggregates the event listeners used on each object into a single list.
-   * @param {Array.<string>} locations Objects to look for attached event listeners.
+   * Aggregates the event listeners used on each element into a single list.
+   * @param {Array.<Element>} nodes List of elements to fetch event listeners for.
    * @return {!Promise<!Array.<Object>>} Resolves to a list of all the event
-   *     listeners found on each object.
+   *     listeners found across the elements.
    */
-  collectListeners(locations) {
-    return locations.reduce((chain, location) => {
+  collectListeners(nodes) {
+    return nodes.reduce((chain, node) => {
       return chain.then(prevArr => {
-        return this.getEventListeners(location).then(results => {
-          return prevArr.concat(results);
-        });
+        return this.getEventListeners(node.element ? node.element.nodeId : node)
+            .then(result => prevArr.concat(result));
```

Comment:
should do `prevArr.push(...result)` as its faster and more conservative in memory allocation.